FEXCore/Source/Interface/IR/Passes/x87StackOptimizationPass.cpp

```diff
@@ -0,0 +1,1148 @@
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+
+      case OP_F80CMPVALUE: {
+        const auto* Op = IROp->C<IROp_F80CmpValue>();
+        const auto& Value = CurrentIR.GetNode(Op->X80Src);
+        auto StackTop = MigrateToSlowPath_IfInvalid();
+        auto StackNode = LoadStackValue(StackTop);
+
+        Ref CmpNode {};
+        if (ReducedPrecisionMode) {
+          CmpNode = IREmit->_FCmp(8, StackNode, Value);
+        } else {
+          CmpNode = IREmit->_F80Cmp(StackNode, Value);
+        }
+        IREmit->ReplaceUsesWithAfter(CodeNode, CmpNode, CodeNode);
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_F80XTRACTSTACK: {
+        auto Value = MigrateToSlowPath_IfInvalid();
+        Ref St0 = LoadStackValue(Value);
+
+        Ref Exp {};
+        Ref Sig {};
+        if (ReducedPrecisionMode) {
+          std::tie(Exp, Sig) = SplitF64SigExp(St0);
+        } else {
+          Exp = IREmit->_F80XTRACT_EXP(St0);
+          Sig = IREmit->_F80XTRACT_SIG(St0);
+        }
+
+        if (SlowPath) {
+          // Write exp to top, update top for a push and set sig at new top.
+          StoreStackValueAtTop_Slow(Exp, false);
+          UpdateTopForPush_Slow();
+          StoreStackValueAtTop_Slow(Sig);
+        } else {
+          StackData.setTop(StackMemberInfo {Exp});
+          StackData.push(StackMemberInfo {Sig});
+        }
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_SYNCSTACKTOSLOW: {
+        // This synchronizes stack values but doesn't necessarily moves us off the FastPath!
+        Ref NewTop = SynchronizeStackValues();
+        IREmit->ReplaceUsesWithAfter(CodeNode, NewTop, CodeNode);
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_STACKFORCESLOW: {
+        MigrateToSlowPathIf(true);
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_INCSTACKTOP:
+      case OP_DECSTACKTOP: {
+        if (SlowPath) {
+          if (IROp->Op == OP_DECSTACKTOP) {
+            UpdateTopForPop_Slow();
+          } else {
+            UpdateTopForPush_Slow();
+          }
+        } else {
+          StackData.rotate(IROp->Op == OP_DECSTACKTOP);
+        }
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_F80ROUNDSTACK: {
+        auto StackMember = MigrateToSlowPath_IfInvalid();
+        Ref St0 = LoadStackValue(StackMember);
+
+        Ref Value {};
+        if (ReducedPrecisionMode) {
+          Value = IREmit->_Vector_FToI(8, 8, St0, Round_Host);
+        } else {
+          Value = IREmit->_F80Round(St0);
+        }
+        StoreStackValue(Value);
+        IREmit->Remove(CodeNode);
+        break;
+      }
+
+      case OP_F80VBSLSTACK: {
+        const auto* Op = IROp->C<IROp_F80VBSLStack>();
+
+        // Multiplies two elements in the stack by offset.
+        auto* VecCond = CurrentIR.GetNode(Op->VectorMask);
+        auto StackOffset1 = Op->SrcStack1;
```

Comment:
No sense making temporaries for these IMHO

Reply:
VecCond can be removed even though I think it aid readability. The others are used in a couple of places so they make sense to stay.